src/app/app.component.ts

```diff
@@ -11,21 +12,28 @@ import { Router } from '@angular/router';
 export class AppComponent implements OnInit {
   title = 'my-app';
   userName = '';
+  loading = false;
 
-  constructor(private authService: AuthenticationService, private router: Router) {}
-
+  constructor(
+    private authService: AuthenticationService, 
+    private courseService:CourseService, 
+    private router: Router
+  ) {}
+  
   ngOnInit(): void {
+    this.courseService.loadingSubject.subscribe(response => {
+      this.loading = response;
+    })
+
     if (this.authService.isAuthenticated()) {
-      this.userName = this.authService.getUserLogin();
-      this.router.navigate(['/courses']);
+      this.authService.getUserLogin().subscribe(response => {
+        this.authService.userSubject.next(JSON.parse(JSON.stringify(response)));
+        this.router.navigate(['/courses']);
```

Comment:
done, made routing `if Auth==false` in Auth Guard, and routing `if Auth==true` in Auth service. 
Because I didn't want Auth Guard forced routed to certain pages, only to check credentials.